lisa/tests/base.py

```diff
@@ -466,6 +469,75 @@ def test_noisy_tasks(self, noise_threshold_pct=None, noise_threshold_ms=None):
         res.add_metric("noisiest task", metric)
 
         return res
+
+    @classmethod
+    #pylint: disable=unused-argument
+    def check_noisy_tasks(cls, noise_threshold_pct=None, noise_threshold_ms=None):
+        """
+        Decorator that applies :meth:`test_noisy_tasks` to the underlying function
+
+        We also expose :meth:`test_noisy_tasks` parameters to the decorated
+        function.
+        """
+        def decorator(func):
+            @cls.test_noisy_tasks.used_events
+            @functools.wraps(func)
+            def wrapper(self, *args,
+                        noise_threshold_pct=noise_threshold_pct,
+                        noise_threshold_ms=noise_threshold_ms,
+                        **kwargs):
+                res = func(self, *args, **kwargs)
+
+                print(noise_threshold_pct)
+
+                noise_res = self.test_noisy_tasks(
+                    noise_threshold_pct, noise_threshold_ms)
+                res.metrics.update(noise_res.metrics)
+
+                if not noise_res:
+                    res.result = Result.UNDECIDED
+
+                return res
+
+            # https://stackoverflow.com/a/33112180
+            # The wrapper has all of `func`'s parameters plus `test_noisy_tasks`',
+            # but since we use `wraps(func)` we'll only get doc/autocompletion for
+            # `func`'s. Expose the extra parameters to the decorated function to
+            # make it more user friendly.
+            func_sig = signature(func)
+            dec_params = signature(cls.check_noisy_tasks).parameters
+            new_params = []
+            # Propagate the default values given to the decorator
+            for name, param in dec_params.items():
+                new_params.append(param.replace(default=locals()[name]))
```

Comment:
is `replace()` needed ? The parameter from `check_noisy_tasks` should already have a `.default` attribute. Accessing the `locals()` seems a bit dangerous since the same name could have been reused/modified.

Reply:
Yeah sadly it is. They do have default values, but it's not the "right ones". If you look at the doc screenshot I shared earlier, that was before this `replace()`. The doc shows that the default value is `None` for both of them, even though the default value specified in the decorator for `test_throughput` is `1`.

To make sense, the default values have to be the ones given to the decorator (they are, after all, the default values for this function). Hence the comment "Propagate the default values *given* to the decorator".

Reply:
Right, in that case could be build a dict with the right variables in them ? The main issue `locals()` brings it that somebody editing the code and looking for usage of a variable before changing it will not find it, and the chances of this being somehow silent are quite high.

Reply:
At that point I might as just get rid of the loop and explicitely create the list. That way we don't get any surprises for the parameter ordering too.